peps/pep-0728.rst

```diff
@@ -141,424 +142,422 @@ in TypeScript achieve this:
         name: str
-        __extra_items__: int
 
     def f(movie: Movie) -> None:
-        del movie["name"]  # Not OK
-        del movie["year"]  # OK
+        del movie["name"]  # Not OK. The value type of 'name' is 'Required[int]'
+        del movie["year"]  # OK. The value type of 'year' is 'NotRequired[int]'
 
 Interaction with ``Unpack``
 ---------------------------
 
-For type checking purposes, ``Unpack[TypedDict]`` with extra items should be
+For type checking purposes, ``Unpack[SomeTypedDict]`` with extra items should be
 treated as its equivalent in regular parameters, and the existing rules for
 function parameters still apply::
 
-    class Movie(TypedDict, closed=True):
+    class Movie(TypedDict, extra_items=int):
         name: str
-        __extra_items__: int
     
     def f(**kwargs: Unpack[Movie]) -> None: ...
 
-    # Should be equivalent to
+    # Should be equivalent to:
     def f(*, name: str, **kwargs: int) -> None: ...
 
-Interaction with PEP 705
-------------------------
+Interaction with Read-only Items
+--------------------------------
 
-When the special ``__extra_items__`` item is annotated with ``ReadOnly[]``, the
-extra items on the TypedDict have the properties of read-only items. This
-interacts with inheritance rules specified in :pep:`PEP 705 <705#Inheritance>`.
+When the ``extra_items`` argument is annotated with the ``ReadOnly[]``
+:term:`type qualifier`, the extra items on the TypedDict have the
+properties of read-only items. This interacts with inheritance rules specified
+in `Read-only Items
+<https://typing.readthedocs.io/en/latest/spec/typeddict.html#id3>`__.
 
-Notably, if the TypedDict type declares ``__extra_items__`` to be read-only, a
-subclass of the TypedDict type may redeclare ``__extra_items__``'s value type or
-additional non-extra items' value type.
+Notably, if the TypedDict type specifies ``extra_items`` to be read-only,
+subclasses of the TypedDict type may redeclare ``extra_items``.
 
 Because a non-closed TypedDict type implicitly allows non-required extra items
-of value type ``ReadOnly[object]``, its subclass can override the special
-``__extra_items__`` with more specific types.
+of value type ``ReadOnly[object]``, its subclass can override the
+``extra_items`` argument with more specific types.
 
 More details are discussed in the later sections.
 
 Inheritance
 -----------
 
-When the TypedDict type is defined as ``closed=False`` (the default),
-``__extra_items__`` should behave and be inherited the same way a regular key
-would. A regular ``__extra_items__`` key can coexist with the special
-``__extra_items__`` and both should be inherited when subclassing.
-
-We assume that ``closed=True`` whenever ``__extra_items__`` is mentioned for the
-rest of this section.
+``extra_items`` is inherited in a similar way as a regular ``key: value_type``
+item. As with the other keys, the `inheritance rules
+<https://typing.readthedocs.io/en/latest/spec/typeddict.html#inheritance>`__
```

Comment:
Ideally for this we should use intersphinx links, not direct URLs.

Reply:
Updated the links where a close label is available. For the remaining ones, looks like there aren't exposed labels available.